src/pods/impl.c

```diff
@@ -0,0 +1,692 @@
+        if((start = strstr(pstore_content, PATTERN1)))
+        {
+                start += strlen(PATTERN1);
+                if ((end = strstr(start, PATTERN2)))
+                {
+                        target = (char *)malloc(end-start+1);
+                        memcpy(target,start,end-start);
+                        target[end-start] = '\0';
+                }
+        }
+        return target;
+}
+
+int get_id_pstore(int id_chk, char *id_type)
+{
+        char pstore_content[BFR_SIZE_256];
+        char acc_id[BFR_SIZE_64];
+        char partner_id[BFR_SIZE_64];
+        char *acc_id_ret = NULL;
+        char *partner_id_ret = NULL;
+        int ret = RETURN_ERR;
+
+        FILE *fptr = NULL;
+        if ((fptr = fopen(OSP_PSTORE_ACCOUNT, "r")) == NULL)
+        {
+                WebcfgError("Error! File %s cannot be opened.",OSP_PSTORE_ACCOUNT);
+                return ret;
+        }
+        memset(pstore_content,0,sizeof(pstore_content));
+        fscanf(fptr, "%[^\n]", pstore_content);
+        WebcfgDebug("Data from the file:\n%s\n", pstore_content);
+        fclose(fptr);
+
+        memset(acc_id,0,sizeof(acc_id));
+        memset(partner_id,0,sizeof(partner_id));
+
+        if((strchr(pstore_content, ',') != NULL) && (strlen(pstore_content) > 0))
+        {
+                sscanf(pstore_content, "%*[^:]%*c%[^,]%*[^:]%*c%[^}]", acc_id, partner_id);
+                acc_id_ret = cutting_delimiters(acc_id, "\"", "\"");
+                partner_id_ret = cutting_delimiters(partner_id, "\"", "\"");
+        }
+        else if(strlen(pstore_content) > 0)
+        {
+                sscanf(pstore_content, "%*[^:]%*c%[^}]", acc_id);
+                acc_id_ret = cutting_delimiters(acc_id, "\"", "\"");
+        }
+        else
+        {
+                WebcfgError("Error!The file is present but is empty!\n");
+                return ret;
+        }
+        if ((id_chk == 1) && (strlen(partner_id) > 0))
+        {
+                cpeabStrncpy(id_type,partner_id_ret,BFR_SIZE_64*sizeof(char));
+                ret = RETURN_OK;
+                return ret;
+        }
+        else
+        {
+                WebcfgError("Partner ID couldnt be parsed\n");
+                return ret;
+        }
+
+        if ((id_chk == 0) && (strlen(acc_id) > 0))
+        {
+                cpeabStrncpy(id_type,acc_id_ret,BFR_SIZE_64*sizeof(char));
+                ret = RETURN_OK;
+                return ret;
+        }
+        else
+        {
+                WebcfgError("Account ID couldnt be parsed\n");
+                return ret;
+        }
+
+}
+char * getPartnerID()
+{
+        int partner_id_chk = 1;
+        char *partner_id = NULL;
+        int ret_val;
+
+  	partner_id = malloc(BFR_SIZE_64*sizeof(char));
+        if (!partner_id)
+        {
+                WebcfgError("%s : partner_id couldnt be assigned with memory\n",__func__);
+                return NULL;
+        }
+        ret_val = get_id_pstore(partner_id_chk,partner_id);
```

Comment:
Instead of passing in memory, why not have `get_id_pstore` just malloc() the memory and return it?